prog/dftb+/lib_math/degeneracy.F90

```diff
@@ -0,0 +1,263 @@
+!--------------------------------------------------------------------------------------------------!
+!  DFTB+: general package for performing fast atomistic simulations                                !
+!  Copyright (C) 2006 - 2020  DFTB+ developers group                                               !
+!                                                                                                  !
+!  See the LICENSE file for terms of usage and distribution.                                       !
+!--------------------------------------------------------------------------------------------------!
+
+#:include 'common.fypp'
+
+module dftbp_degeneracyDetect
```

Comment:
I'm not too keen starting with camelCase in module names. Also degeneracyDectect is more than the 15 character limit from the style guide. Maybe use just degeneracy or detectdeg here?